WeatherService/Presentation/WeatherDataParserFactory.cs

```diff
@@ -0,0 +1,19 @@
+﻿
+using WeatherService.Parsers;
+
+namespace WeatherService.Presentation
+{
+    public class WeatherDataParserFactory
+    {
+        public IWeatherDataParser? DetectDataFormat(string inputData)
```

Comment:
Also, this one should be part of the services folder not the presentation layer. 
```suggestion
        public IWeatherDataParser? Create(string inputData)
```